Sanitizer + omp backend = undefined reference during linking

Hi,

I just came accros this weirdness, and after some cleaning, this simple main create an undefined symbol during linking 

```c++
#include <sycl/sycl.hpp>

int main() {
   
    sycl::queue queue;

    queue.submit([&] (sycl::handler& cgh) {
        
        cgh.parallel_for(sycl::nd_range<1>(32 , 32),[=] (sycl::nd_item<1> item) {

            size_t local_id = item.get_local_id(0);
            size_t global_id = item.get_global_id(0);
            
            item.barrier(sycl::access::fence_space::local_space);

        });
    });
    

    return 0;
}
```

The following compile command : 

```bash
./sycl_cpl/hipSYCL/bin/syclcc -std=c++17 -g -fsanitize=address\
    --hipsycl-config-file=./sycl_cpl/hipSYCL/etc/hipSYCL/syclcc.json\
    --hipsycl-targets='omp' --hipsycl-platform=cpu  test.cpp -o out
```

gives:

```
/usr/bin/ld: /tmp/test-61dbfd.o: in function `void hipsycl::glue::omp_dispatch::iterate_nd_range_omp<1, main::$_0::operator()(hipsycl::sycl::handler&) const::{lambda(hipsycl::sycl::nd_item<1>)#1}>(main::$_0::operator()(hipsycl::sycl::handler&) const::{lambda(hipsycl::sycl::nd_item<1>)#1}, hipsycl::sycl::id<1> const&&, hipsycl::sycl::range<1>, hipsycl::sycl::range<1>, hipsycl::sycl::id<1>, unsigned long, void*, std::function<void ()>&)':
......./hipSYCL/bin/../include/hipSYCL/glue/omp/omp_kernel_launcher.hpp:235: undefined reference to `__hipsycl_local_id_x'
/usr/bin/ld: ......./hipSYCL/bin/../include/hipSYCL/glue/omp/omp_kernel_launcher.hpp:235: undefined reference to `__hipsycl_local_id_x'
```

removing the ```-fsanitize=address``` solve the issue.

Note : i'm using the default config for hipsycl (commit 56b6c1f4198aced999a4199b272790a6e15adabe) : 

```
{
  "version-major" : "0",
  "version-minor" : "9",
  "version-patch" : "3",
  "plugin-llvm-version-major" : "14",
  "plugin-with-cpu-acceleration" : "true",
  "default-clang"     : "/usr/bin/clang++",
  "default-nvcxx"     : "NVCXX_COMPILER-NOTFOUND",
  "default-platform"  : "cuda",
  "default-cuda-path" : "/opt/cuda",
  "default-gpu-arch"  : "",
  "default-cpu-cxx"   : "/usr/bin/c++",
  "default-rocm-path" : "/opt/rocm",
  "default-use-bootstrap-mode" : "false",
  "default-is-dryrun" : "false",
  "default-use-accelerated-cpu" : "true",
  "default-clang-include-path" : "/usr/lib/clang/14.0.6/include/..",
  "default-sequential-link-line" : "-L/usr/lib -lboost_context -lboost_fiber -Wl,-rpath=/usr/lib",
  "default-sequential-cxx-flags" : "-I/usr/include -D_ENABLE_EXTENDED_ALIGNED_STORAGE",
  "default-omp-link-line" : "-L/usr/lib -lboost_context -lboost_fiber -Wl,-rpath=/usr/lib -fopenmp",
  "default-omp-cxx-flags" : "-I/usr/include -fopenmp -D_ENABLE_EXTENDED_ALIGNED_STORAGE",
  "default-rocm-link-line" : "-Wl,-rpath=$HIPSYCL_ROCM_PATH/lib -Wl,-rpath=$HIPSYCL_ROCM_PATH/hip/lib -L/opt/rocm/lib -L/opt/rocm/hip/lib -lamdhip64",
  "default-rocm-cxx-flags" : "-isystem $HIPSYCL_PATH/include/hipSYCL/std/hiplike -isystem /usr/lib/clang/14.0.6/include/.. -U__FLOAT128__ -U__SIZEOF_FLOAT128__ -I$HIPSYCL_ROCM_PATH/include -I$HIPSYCL_ROCM_PATH/include --rocm-device-lib-path=$HIPSYCL_ROCM_PATH/amdgcn/bitcode --rocm-path=$HIPSYCL_ROCM_PATH -fhip-new-launch-api -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false -D__HIP_ROCclr__",
  "default-cuda-link-line" : "-Wl,-rpath=$HIPSYCL_CUDA_LIB_PATH -L$HIPSYCL_CUDA_LIB_PATH -lcudart",
  "default-cuda-cxx-flags" : "-U__FLOAT128__ -U__SIZEOF_FLOAT128__ -isystem $HIPSYCL_PATH/include/hipSYCL/std/hiplike",
  "default-is-explicit-multipass" : "false",
  "default-save-temps" : "false"
}
```

--- a/src/compiler/cbs/RemoveBarrierCalls.cpp
+++ b/src/compiler/cbs/RemoveBarrierCalls.cpp
@@ -27,16 +27,30 @@
 #include "hipSYCL/compiler/cbs/SplitterAnnotationAnalysis.hpp"
 
 namespace {
+
 bool deleteGlobalVariable(llvm::Module *M, llvm::StringRef VarName) {
   if (auto *GV = M->getGlobalVariable(VarName)) {
-    if (GV->getNumUses() == 0) {
+    llvm::SmallVector<llvm::Instruction *, 8> WL;
+    for (auto U : GV->users())
+      if (auto LI = llvm::dyn_cast<llvm::LoadInst>(U); LI && LI->user_empty())
+        WL.push_back(LI);
+    for (auto *LI : WL)
+      LI->eraseFromParent();
+
+    if (GV->getNumUses() == 0 ||
+        std::none_of(GV->user_begin(), GV->user_end(), [GV](llvm::User *U) { return U != GV; })) {
       HIPSYCL_DEBUG_INFO << "[RemoveBarrierCalls] Clean-up global variable " << *GV << "\n";
       GV->eraseFromParent();
       return true;
     }
+    HIPSYCL_DEBUG_INFO << "[RemoveBarrierCalls] Global variable still in use " << VarName << "\n";
+    for (auto *U : GV->users()) {
+      HIPSYCL_DEBUG_INFO << "[RemoveBarrierCalls] >>> " << *U << "\n";
+    }
   }
   return false;
 }
+
 bool removeBarrierCalls(llvm::Function &F, hipsycl::compiler::SplitterAnnotationInfo &SAA) {
   if (!SAA.isKernelFunc(&F))
     return false;

